src/output_writers.jl

```diff
@@ -1,14 +1,49 @@
 import JLD
 
+"A type for writing checkpoints."
 struct Checkpointer <: OutputWriter
     dir::AbstractString
-    filename_prefix::AbstractString
-    output_frequency::Int
+    prefix::AbstractString
+    frequency::Int
```

Comment:
I think `filename_prefix` and `output_frequency` make the code clearer and more self-documenting.

It's obvious what they do in this context but if you have a field writer called `fw` then `fw.prefix` and `fw.frequency` seem somewhat obscure.

Reply:
That's a style issue and I'm fine if you change it, though I do prefer my implementation. I guess I am proposing that an `OutputWriter` has one job (writing output); so that it is evident that all of its fields refer to that task.

Reply:
Thinking about this more, I've decided I'm fine with the verbosity of things like `output_frequency` and `diagnostic_frequency` if it makes the code clearer.